File: 프로그래머스/1/12982. 예산/예산.js
function solution(d, budget) {
    d.sort((a,b)=>a-b);
    let result = 0;
    for(let i = 0; i < d.length; i++) {
        if(budget >= d[i]) {
            result++;
            budget-=d[i];
        }
    }
    return result;
}

File: 백준/Silver/2512. 예산/예산.py
import sys

def s_input():
    return sys.stdin.readline().strip()

N = int(s_input())
moneys = list(map(int, s_input().split(" ")))
total_money = int(s_input())
moneys.sort()
result = 0
for i in range(len(moneys)):
    max_money = total_money // (N - i)
    if max_money >= moneys[i]:
        total_money -= moneys[i]
        result = moneys[i]
    else:
        result = max_money
        break
print(result)